Remove requires and references to OrderedHash

okuramasafumi opened the pull request:
### Summary

OrderedHash is deprecated but there are some requires and references to OrderedHash, which might be confusing.
As described in #22681, OrderedHash is internal only so references in the docs should be removed.

kamipo commented:
See #22681.

Removing `require "active_support/ordered_hash"` and the reference in the doc is fine though.

okuramasafumi committed:
Remove some references to OrderedHash

OrderedHash is deprecated but there are some requires and references
to OrderedHash, which might be confusing.
As described in
https://github.com/rails/rails/issues/22681#issuecomment-166059717
OrderedHash is internal only so references in the docs should be
removed.

okuramasafumi commented:
@kamipo Thank you for pointing it out!
